pkg/test/nodes.go

```diff
@@ -67,15 +68,23 @@ func Node(overrides ...NodeOptions) *v1.Node {
 	}
 }
 
+// this check is done for tests that assume the nodeclaim is past registration but will not go through reconciliation
 func NodeClaimLinkedNode(nodeClaim *v1beta1.NodeClaim) *v1.Node {
+	taints := lo.Reject(nodeClaim.Spec.StartupTaints, func(t v1.Taint, _ int) bool {
```

Comment:
Do we expect the unregistered taint to be in the startup taints anymore?

Reply:
Nope.